### DISCUSSION TITLE
Should created users and settings persist through docker image recreations?

### DISCUSSION THREAD

**Discussion by shruggle** [filebrowser/filebrowser#2115] [Q&A]
Hi, I'm fairly new to using docker so apologies if this is a dumb question. 

I've been messing around with the docker filebrowser image and I noticed that if I remove the image and pull it again I will have to recreate the users and change the settings I had changed.

Honestly I'm mostly just curious if I have set something up wrong, so no worries if that is indeed intended behavior.

Here is what I have in my docker-compose.yml:

> filebrowser:
>         volumes:
>             - '/home/user/Downloads:/srv'
>             - '/docker/appdata/filebrowser/filebrowser.db:/database.db'
>             - '/docker/appdata/filebrowser/settings.json:/settings.json'
>         environment:
>             - PUID=1000
>             - PGID=1000
>         ports:
>             - '8080:80'
>         image: 'filebrowser/filebrowser:s6'
>         restart: unless-stopped

**Comment by ramiresviana** [comment]
Using bind mounts can be used to persist data from containers, preventing to lose the data when the container is restarted or re-created. Your configuration is mounting the files from `/docker/appdata/filebrowser/` into the container, allowing it to use the existing database. Since you're providing a config file for Filebrowser, you must ensure that the path used for the the database file is accessible. Checking the container logs can provide useful information such errors and configuration. Take a look at the Docker documentation for more info: https://docs.docker.com/
